.github/workflows/ci.yaml

```diff
@@ -80,6 +80,35 @@ jobs:
         run: sudo make build
       - name: test-e2e
         run: sudo -E make test-e2e-for-awslogs # containerd interaction requires sudo and aws cloudwatch interaction requires passing env vars
+  e2e-tests-for-fluentd:
+    strategy:
+      fail-fast: false
+      matrix:
+        go: [ '1.20', '1.21' ]
+        os: [ ubuntu-latest ] # TODO: Add Windows e2e tests: https://github.com/aws/shim-loggers-for-containerd/issues/68
+    name: E2E tests / fluentd / ${{ matrix.os }} / Go ${{ matrix.go }}
+    runs-on: ${{ matrix.os }}
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
+          cache: false
+      - name: install and start containerd
+        shell: bash
+        run: sudo scripts/install-containerd
+      - name: start fluentd local endpoint
+        shell: bash
+        run: | # Fluentd container is not using root user so need 777 to make it writable. https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#docker-container-filesystem
+          sudo mkdir -m 777 $GITHUB_WORKSPACE/fluentd-logs
+          docker run -d -p 24224:24224 -p 24224:24224/udp -v $GITHUB_WORKSPACE/fluentd-logs:/fluentd/log public.ecr.aws/docker/library/fluentd:v1.16-debian-1
```

Comment:
should some of this variable / parameterized?

Reply:
Revised.